src/components/Router/index.js

```diff
@@ -117,8 +117,10 @@ class CrossDeviceMobileRouter extends Component {
     actions.acceptTerms()
   }
 
-  setError = () =>
-    this.setState({crossDeviceError: true, loading: false})
+  setError = (error) => {
```

Comment:
hmm, might more semantic to put `GENERIC_CLIENT_ERROR` as the default of `error`, otherwise `setError(false)` will turn the generic error and it won't be very clear.